pkg/transparentproxy/kubernetes/kubernetes.go

```diff
@@ -0,0 +1,119 @@
+}
+
+func NewPodRedirectForPod(pod *kube_core.Pod) (*PodRedirect, error) {
+	var err error
+	podRedirect := &PodRedirect{}
+
+	podRedirect.BuiltinDNSEnabled, _, err = metadata.Annotations(pod.Annotations).GetEnabled(metadata.KumaBuiltinDNS)
+	if err != nil {
+		return nil, err
+	}
+
+	podRedirect.BuiltinDNSPort, _, err = metadata.Annotations(pod.Annotations).GetUint32(metadata.KumaBuiltinDNSPort)
+	if err != nil {
+		return nil, err
+	}
+
+	podRedirect.ExcludeOutboundPorts, _ = metadata.Annotations(pod.Annotations).GetString(metadata.KumaTrafficExcludeOutboundPorts)
+
+	podRedirect.RedirectPortOutbound, _, err = metadata.Annotations(pod.Annotations).GetUint32(metadata.KumaTransparentProxyingOutboundPortAnnotation)
+	if err != nil {
+		return nil, err
+	}
+
+	podRedirect.RedirectInbound = true
+	enabled, exist, err := metadata.Annotations(pod.Annotations).GetEnabled(metadata.KumaGatewayAnnotation)
+	if err != nil {
+		return nil, err
+	}
+	if exist && enabled {
+		podRedirect.RedirectInbound = false
+	}
+
+	podRedirect.ExcludeInboundPorts, _ = metadata.Annotations(pod.Annotations).GetString(metadata.KumaTrafficExcludeInboundPorts)
+
+	podRedirect.RedirectPortInbound, _, err = metadata.Annotations(pod.Annotations).GetUint32(metadata.KumaTransparentProxyingInboundPortAnnotation)
+	if err != nil {
+		return nil, err
+	}
+
+	podRedirect.RedirectPortInboundV6, _, err = metadata.Annotations(pod.Annotations).GetUint32(metadata.KumaTransparentProxyingInboundPortAnnotationV6)
+	if err != nil {
+		return nil, err
+	}
+
+	podRedirect.UID, _ = metadata.Annotations(pod.Annotations).GetString(metadata.KumaSidecarUID)
+
+	return podRedirect, nil
+}
+
+func (pr *PodRedirect) AsTransparentProxyConfig() *config.TransparentProxyConfig {
+	return &config.TransparentProxyConfig{
+		DryRun:                 false,
+		Verbose:                false,
+		RedirectPortOutBound:   fmt.Sprintf("%d", pr.RedirectPortOutbound),
+		RedirectInBound:        pr.RedirectInbound,
+		RedirectPortInBound:    fmt.Sprintf("%d", pr.RedirectPortInbound),
+		RedirectPortInBoundV6:  fmt.Sprintf("%d", pr.RedirectPortInboundV6),
+		ExcludeInboundPorts:    pr.ExcludeInboundPorts,
+		ExcludeOutboundPorts:   pr.ExcludeOutboundPorts,
+		UID:                    pr.UID,
+		GID:                    pr.UID, // TODO: shall we have a separate annotation here?
+		RedirectDNS:            pr.BuiltinDNSEnabled,
+		RedirectAllDNSTraffic:  false,
+		AgentDNSListenerPort:   fmt.Sprintf("%d", pr.BuiltinDNSPort),
+		DNSUpstreamTargetChain: "",
+	}
+}
+
+func (pr *PodRedirect) AsKumactlCommandLine() []string {
+	dnsArg := []string{
+		"--skip-resolv-conf",
+	}
+
+	if pr.BuiltinDNSEnabled {
+		dnsArg = append(dnsArg,
+			"--redirect-all-dns-traffic",
+			"--redirect-dns-port", strconv.FormatInt(int64(pr.BuiltinDNSPort), 10),
+		)
+	}
+
+	return append([]string{
+		"--redirect-outbound-port",
+		fmt.Sprintf("%d", pr.RedirectPortOutbound),
+		"--redirect-inbound=" + fmt.Sprintf("%t", pr.RedirectInbound),
```

Comment:
nit: why this one is with `--arg=value` where other arguments are `--arg value`?

Reply:
This one is bool argument, this is the only way to assign a value to it.

Reply:
we could add `--redirect-inbound` only if `pr.RedirectInbound == true` but probably not worth it.